README.md

```diff
@@ -37,185 +41,198 @@ Simple parsing for node.js.
+Add a script to `scripts` in `package.json` that runs the command above if you only have a locally installed copy of nearley.
+
+- Create a parser and feed it data:
+
+```js
+const nearley = require("nearley");
+const grammar = require("./grammar");
+
+// Create a Parser object from our grammar.
+const parser = new nearley.Parser(nearley.Grammar.fromCompiled(grammar));
+
+// Parse something.
+parser.feed("foo\n");
 
-```html
-<script src="nearley.js"></script>
-<script src="my-generated-grammar.js"></script>
+// parser.results is an array of possible parsings.
+// It's empty if we're unable to parse the input.
+// If the input can be interpreted in multiple ways, parser.results will contain all of them.
+console.log(parser.results); // [[[[ "foo" ],"\n" ]]]
 ```
 
+See below for detailed API and grammar syntax specification.
 
-Parser specification
---------------------
+## CLI
 
-This is a basic overview of nearley syntax and usage. For an advanced
-styleguide, see [this file](how-to-grammar-good.md).
+Use `--help` with any of these commands to see available options.
 
-A parser consists of several *nonterminals*, which are constructions in a
-language. A nonterminal is made up of a series of either other nonterminals or
-strings. In nearley, you define a nonterminal by giving its name and its
-expansions.
+- `nearleyc` compiles grammar files to JavaScript.
+- `nearley-test` lets you quickly test a grammar against some input and see the
+  results. It also lets you explore the internal state of nearley's Earley
+  table, in case you find that interesting.
+- `nearley-unparse` inverts a parser into a generator, allowing you to create
+  random strings that match your grammar.
+- `nearley-railroad` generates pretty railroad diagrams from your parser. This
+  is mainly helpful for creating documentation, as (for example) on json.org.
+
+## Parser specification
+
+Let's explore the building blocks of a nearley parser.
+
+### Terminals, nonterminals, rules
+
+- A *terminal* is a string or a token. E.g. keyword `"if"` is a terminal.
+- A *nonterminal* is a combination of terminals and other nonterminals. E.g. an if statement defined as `"if" condition statement` is a nonteminal.
+- A *rule* (or production rule) is a definition of a nonterminal. E.g. `"if" condition statement` is the rule according to which the if statement nonterminal is parsed.
 
-Strings are the *terminals*, which match those string literals (specified as
-JSON-compatible strings).
+The first nonterminal of the grammar is the one the whole input must match. With the following grammar, nearley will try to parse text as `expression`.
 
-The following grammar matches a number, a plus sign, and another number:
+```js
+expression -> number "+" number
+number -> [0-9]:+
+```
 
-    expression -> number "+" number
-    
-Anything from a `#` to the end of a line is ignored as a comment:
+A nonterminal can have multiple alternative rules, separated by vertical bars (`|`):
```

Comment:
I feel weird about how most of the examples here use `"scannerless"` mode, while we in theory strongly encourage using lexers. :P